packages/lisk-transaction-pool/src/transaction_pool.ts

```diff
@@ -40,62 +40,136 @@ interface Queues {
+		return this._queues.ready.dequeueUntil(
+			queueCheckers.returnTrueUntilLimit(limit),
+		);
+	}
 
-	public onRoundRollback(delegates: ReadonlyArray<string>): void {}
+	public onDeleteBlock(block: Block): void {
+		const { received, validated, ...otherQueues } = this._queues;
 
-	public verifyTransaction(): void {}
+		// Move transactions from the verified, pending and ready queues to the validated queue where account was a receipient in the delete block
+		const transactionsToAffectedAccounts = this.removeTransactionsFromQueues(
+			otherQueues,
+			queueCheckers.checkTransactionForRecipientId(block.transactions),
+		);
 
-	private existsInTransactionPool(transaction: Transaction): boolean {}
+		this._queues.validated.enqueueMany(transactionsToAffectedAccounts);
+		// Add transactions to the verfied queue which were included in the deleted block
+		this._queues.verified.enqueueMany(block.transactions);
+	}
 
-	private expireTransactions(): void {}
+	public onNewBlock(block: Block): void {
+		// Remove transactions in the transaction pool which were included in the new block
+		this.removeTransactionsFromQueues(
+			this._queues,
+			queueCheckers.checkTransactionForId(block.transactions),
+		);
+
+		const { received, validated, ...otherQueues } = this._queues;
+		// Remove transactions from the verified, pending and ready queues which were sent from the accounts in the new block
+		const transactionsFromAffectedAccounts = this.removeTransactionsFromQueues(
+			otherQueues,
+			queueCheckers.checkTransactionForSenderPublicKey(block.transactions),
+		);
+
+		// Remove all transactions from the verified, pending and ready queues if they are of a type which includes unique data and that type is included in the block
+		// TODO: remove the condition for checking `containsUniqueData` exists, because it should always exist
+		const blockTransactionsWithUniqueData = block.transactions.filter(
+			(transaction: Transaction) =>
+				transaction.containsUniqueData && transaction.containsUniqueData(),
+		);
+		const transactionsOfTypesWithUniqueData = this.removeTransactionsFromQueues(
+			otherQueues,
+			queueCheckers.checkTransactionForTypes(blockTransactionsWithUniqueData),
+		);
+
+		// Add transactions which need to be reverified to the validated queue
+		this._queues.validated.enqueueMany([
+			...transactionsFromAffectedAccounts,
+			...transactionsOfTypesWithUniqueData,
+		]);
+	}
 
-	private processVerifiedTransactions(): void {}
+	public onRoundRollback(delegates: ReadonlyArray<string>): void {
+		// Move transactions from the verified, pending and ready queues to the validated queue which were sent from delegate accounts
+		const { received, validated, ...otherQueues } = this._queues;
+		const senderProperty: queueCheckers.TransactionFilterableKeys =
+			'senderPublicKey';
+		const transactionsFromAffectedAccounts = this.removeTransactionsFromQueues(
+			otherQueues,
+			queueCheckers.checkTransactionPropertyForValues(
```

Comment:
Because checkTransactionForSenderPublicKey expects transaction objects to be passed, where as here we don't  have transaction objects, and only a list of publicKeys.